src/app.rs

```diff
@@ -289,6 +291,19 @@ where
             states.stop(StateData::new(world, &mut self.data));
         }
 
+        // Read the Trans queue and apply changes.
+        {
+            let trans = {
+                let mut v = self.world.write_resource::<TransQueue<T, E>>();
+                let x = v.drain(..).collect::<VecDeque<_>>();
+                x
+            };
+            let states = &mut self.states;
```

Comment:
Its just the way it was written before.
We have some people who like to separate the borrow and the usage.
states.transition at line 303 requires a mutable reference, so here it is create before the loop instead of inside (doesn't really matter, the compiler would have optimized it to be outside of the loop anyway)

Reply:
No, the reason it was done that way is to tell the compiler it's safe to have mutable references to multiple parts of self. If you don't do this you will get a borrow error between self.states and self.world.

Reply:
ah okay